app-preview/src/main/java/com/mapbox/navigation/examples/preview/dropinui/RequestRouteWithNavigationViewActivity.kt

```diff
@@ -0,0 +1,154 @@
+        }
+
+        override fun onNewRawLocation(rawLocation: Location) {
+            // no impl
+        }
+    }
+
+    /**
+     * Notifies with attach and detach events on [MapView]
+     */
+    private val mapViewObserver = object : MapViewObserver() {
+        override fun onAttached(mapView: MapView) {
+            mapView.gestures.addOnMapLongClickListener(
+                this@RequestRouteWithNavigationViewActivity
+            )
+        }
+
+        override fun onDetached(mapView: MapView) {
+            mapView.gestures.removeOnMapLongClickListener(
+                this@RequestRouteWithNavigationViewActivity
+            )
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = MapboxActivityRequestRouteNavigationViewBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        // Set to false if you want to handle map long click listener in the app
+        binding.navigationView.customizeViewOptions {
+            enableMapLongClickIntercept = false
+        }
+
+        binding.navigationView.registerMapObserver(mapViewObserver)
+        MapboxNavigationApp.current()?.registerLocationObserver(locationObserver)
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        binding.navigationView.unregisterMapObserver(mapViewObserver)
+        MapboxNavigationApp.current()?.unregisterLocationObserver(locationObserver)
+    }
+
+    override fun onMapLongClick(point: Point): Boolean {
+        ifNonNull(lastLocation) {
+            requestRoutes(Point.fromLngLat(it.longitude, it.latitude), point)
+        }
+        return false
+    }
+
+    private fun requestRoutes(origin: Point, destination: Point) {
+        MapboxNavigationApp.current()!!.requestRoutes(
+            routeOptions = RouteOptions
+                .builder()
+                .applyDefaultNavigationOptions()
+                .applyLanguageAndVoiceUnitOptions(this)
+                .coordinatesList(listOf(origin, destination))
+                .alternatives(true)
+                .build(),
+            callback = object : NavigationRouterCallback {
+                override fun onCanceled(routeOptions: RouteOptions, routerOrigin: RouterOrigin) {
+                    // no impl
+                }
+
+                override fun onFailure(reasons: List<RouterFailure>, routeOptions: RouteOptions) {
+                    // no impl
+                }
+
+                override fun onRoutesReady(
+                    routes: List<NavigationRoute>,
+                    routerOrigin: RouterOrigin
+                ) {
+                    binding.navigationView.api.enableReplaySession()
+                    // Sets the destination.
+                    binding.navigationView.api.setDestination(destination)
+                    // Sets routes to preview state.
+                    binding.navigationView.api.setPreviewRoutes(routes)
```

Comment:
do we need this if we go to the AG state immediately?

Reply:
You go into route preview state by pressing the back button. Without this call there won't be any routes in this state.